src/modules/i18n/containers/__tests__/I18nProviderContainer.spec.js

```diff
@@ -2,18 +2,41 @@
 
 import configureMockStore from 'redux-mock-store'
 import { Provider } from 'react-redux'
-import I18nProviderContainer from '../I18nProviderContainer'
 import * as React from 'react'
 import waitForExpect from 'wait-for-expect'
 import { render } from '@testing-library/react-native'
+import TestRenderer from 'react-test-renderer'
+import typeof I18nProviderContainer from '../I18nProviderContainer'
 
 jest.mock('@react-native-community/async-storage')
 jest.mock('../../../i18n/LanguageDetector')
 
 const mockStore = configureMockStore()
 
 describe('I18nProviderContainer', () => {
-  it('should pass the right props', async () => {
+  beforeEach(() => {
+    jest.resetModules()
+  })
+
+  it('should pass props to inner component', () => {
+    const I18nProviderMock = () => null
+    jest.doMock('../../components/I18nProvider', () => I18nProviderMock)
+    const I18nProviderContainerMock: I18nProviderContainer = require('../I18nProviderContainer').default
+
+    const store = mockStore({})
+    const rendered = TestRenderer.create(<Provider store={store}>
+      <I18nProviderContainerMock />
+    </Provider>)
+    const instance = rendered.root.findByType(I18nProviderMock)
+    expect(instance.props).toEqual({
+      setContentLanguage: expect.anything()
```

Comment:
one should test, that setContentLanguage dispatches an action, since this is part of the container logic
(call it, then check store.getActions)

Reply:
It is tested indirectly by the test below. but I added an explicit test